src/RestQuery.js

```diff
@@ -752,22 +752,20 @@ RestQuery.prototype.handleInclude = function() {
     return;
   }
 
-  var pathResponse = includePath(
-    this.config,
-    this.auth,
-    this.response,
-    this.include[0],
-    this.restOptions
-  );
-  if (pathResponse.then) {
-    return pathResponse.then(() => {
-      this.include = this.include.slice(1);
-      return this.handleInclude();
-    });
-  } else if (this.include.length > 0) {
-    this.include = this.include.slice(1);
-    return this.handleInclude();
-  }
+  // Construct a graph of promises, ensuring that we don't try to load
+  // any path until it's prefix has finished loading.
+  const promises = { '': Promise.resolve() };
+  this.include.forEach(path => {
```

Comment:
do you think its worth adding some kind of concurrency limiter so we're sure we're not doing more that 10 ish concurrent lookups?